src/Microsoft.Health.Dicom.Core/Features/CustomTag/AddCustomTagHandler.cs

```diff
@@ -13,17 +13,17 @@ namespace Microsoft.Health.Dicom.Core.Features.CustomTag
 {
     public class AddCustomTagHandler : IRequestHandler<AddCustomTagRequest, AddCustomTagResponse>
```

Comment:
>AddCustomTagHandler [](start = 17, length = 19)

What are the various responsibilities of the different layers here? It looks like the `IRequestHandler<TIn, TOut>` interface is for `MediatR`, but what is the semantic difference between the `IAddCustomTagSerice` and the `ICustomTagService`. It seems like if there is necessary input/output shimming, it can be done within the handler directly. However, I can imagine keeping the service and store separate if the Add action itself has specific actions that must be taken (and is worth encapsulating in its own entity)

Reply:
I've split this so we have separate services for the actions on custom tag. This is primarily because I want to keep the class responsibility small and easy to read. Add I imagine will get complex, as will remove!

---
In reply to: [564841468](https://github.com/microsoft/dicom-server/pull/501#discussion_r564841468) [](ancestors = 564841468)